### DISCUSSION TITLE
New settings page - mockup

### DISCUSSION THREAD

**Discussion by Secret-chest** [ScratchAddons/ScratchAddons#5893] [Discussion]
@WorldLanguages I suppose I should ping you, since you gave me a rocket emoji

Open it in a new tab, make sure to have 100% zoom and go fullscreen to get a better idea

I didn't add more cards, but all lists scroll vertically

#### Settings page - default view
![scratchaddons_Page 1](https://user-images.githubusercontent.com/74449186/230713734-000fb00b-02d4-4fbb-b4af-8c5e0a8bf349.png)
#### Settings page - compact view
![scratchaddons_Page 2](https://user-images.githubusercontent.com/74449186/230713754-13779c44-2840-432a-8640-3b5cf5be2fb7.png)

### "Mobile" / popup
|                                                                                                                                |                                                                                                                                |                                                                                                                                |
|--------------------------------------------------------------------------------------------------------------------------------|--------------------------------------------------------------------------------------------------------------------------------|--------------------------------------------------------------------------------------------------------------------------------|
| **Settings page - large view**                                                                                                 | **Settings page - medium/default view**                                                                                        | **Settings page - compact/list view**                                                                                          |
| ![scratchaddons_Page 3](https://user-images.githubusercontent.com/74449186/230713813-c1cb0a1b-9a33-45af-8a94-ff63c56da1e5.png) | ![scratchaddons_Page 4](https://user-images.githubusercontent.com/74449186/230713821-b0b8be6b-6cf6-46be-8056-e4ebc4145912.png) | ![scratchaddons_Page 5](https://user-images.githubusercontent.com/74449186/230713837-a1433e37-2d2d-41c2-a49b-719709e11440.png) |
| **Search bar in settings page**                                                                                                | **Addon settings in settings page**                                                                                            | **Sidenav (collapsible)**                                                                                                      |
| ![scratchaddons_Page 6](https://user-images.githubusercontent.com/74449186/230713861-e6d02541-f588-4033-a306-79452a83edd8.png) | ![scratchaddons_Page 7](https://user-images.githubusercontent.com/74449186/230713874-52993a96-bd15-452e-8f9f-2d41e7dbd6d3.png) | ![scratchaddons_Page 8](https://user-images.githubusercontent.com/74449186/230713890-1ba38dd9-7dfe-4d99-8272-01c83f131d7f.png) |
| **Messaging page**                                                                                                | ** **                                                                                            | ** **                                                                                                      |
| ![scratchaddons_Page 9](https://user-images.githubusercontent.com/74449186/230717444-58e7d12a-7ff4-48d2-bad1-eb49c9a6620a.png) | | |

To open addon settings on mobile / popup, click the addon card. To close, click outside of the panel.

Yes it's material design, but we're talking about the layout, not style

### Main Changes
* material design :)
* changed font to roboto
* made the background a cooler grey and not completely dark
* top bar is now grey to allow for better focus (only in dark mode)
* sidenav: provides access to popup tabs as well as links (feedback, contribute etc.)
    * on mobile/popup it is collapsible
* new card layout:
    * no more expand / collapse
    * shows addon image (thanks [placekitten](https://placekitten.com)!)
    * shows a wiki button next to the switch
    * still shows description + notices
* settings are now accessed in the setting panel, only for the currently selected addon:
    * on desktop the panel is to the right and always.visible
    * on mobile/popup it opens in a bottom sheet over the page, only after clicking on an addon. To close it, click outside
    * also shows description and notes
    * addons can still be toggled directly from their card
* setting names are no longer uppercase
* replaced categories and groups with filters
* More Settings is now called Extension Settings
* icons are now filled [Community Material Icons](https://pictogrammers.com/library/mdi/) for more clarity (also available on [iconify](icon-sets.iconify.io/mdi)
* view options:
    * list / compact: small cards, shows only name, tags on desktop only, help button, switch
    * medium (mobile/popup-only, default): same as large, but without images
    * large (default on desktop): shows everything (image, name, tags, help button, switch, description, notes and credits)
        * on desktop image is to the left and tags are in the header
        * on mobile/popup it is all stacked

**Comment by Secret-chest** [comment]
The 3 search bar icons are:
* view
    * (default) list (shown)
    * compact list (no images and descriptions, similar to @DNin01 's design)
    * grid (uses cards with large images and descriptions)
* sort
    * relevance (the one we have now)
    * relative popularity (based on usage data from feedback page)
    * alphabetical
* filters - replaces categories, groups, subgroups, uses tags (they still exist, but not shown in the example)
    * applies to: editor, project player, project page, studio, website, forums, etc.
    * recommandation: recommended, featured, all
    * type: addition, tweak, fix, theme
    * possibly more

**Comment by Secret-chest** [comment]
I'm open to implement at least part of this, but first let's reach a consensus

**Reply by WorldLanguages** [reply]
Before implementation, we'd probably have to discuss whether to continue using Vue v1 or use a build step.
Please let's avoid discussing the above within this discussion.

**Reply by Secret-chest** [reply]
> Before implementation, we'd probably have to discuss whether to continue using Vue v1 or use a build step. Please let's avoid discussing the above within this discussion.

Fair enough. Let's keep this discussion about design.

**Comment by Secret-chest** [comment]
Help button would go to the addon's wiki page

**Comment by Secret-chest** [comment]
"My Stuff" is a hypothetical new popup tab that would act as a quick project manager

**Comment by Samq64** [comment]
I like the idea of tags, but I also categories need to stay. Maybe not all of them, but at least editor, website and themes.

**Reply by Secret-chest** [reply]
But themes can be for a specific page. This is why filters are better (they would use a special search syntax)

**Reply by WorldLanguages** [reply]
I am not attracted to the idea of removing a one-click button that lets users filter by editor addons vs website addons. Filtering with special search syntax can be an addition instead of completely replacing the current sidebar UI.

**Reply by Secret-chest** [reply]
> I am not attracted to the idea of removing a one-click button that lets users filter by editor addons vs website addons. Filtering with special search syntax can be an addition instead of completely replacing the current sidebar UI.

They still have a GUI

**Reply by WorldLanguages** [reply]
Where is it? Is it hidden inside the filters?

**Reply by Secret-chest** [reply]
> Where is it? Is it hidden inside the filters?

The "funnel" icon to the right of the search bar

**Reply by DNin01** [reply]
I think having quickly selectable categories to filter addons is more user-friendly than interchangeable filters. It doesn't need to be super advanced - we only have about 150 addons.

**Reply by Secret-chest** [reply]
> I think having quickly selectable categories to filter addons is more user-friendly than interchangeable filters. It doesn't need to be super advanced - we only have about 150 addons.

I explained it in a comment

**Reply by WorldLanguages** [reply]
> The "funnel" icon to the right of the search bar

That's not very noticeable.

**Reply by Secret-chest** [reply]
> > The "funnel" icon to the right of the search bar
> 
> That's not very noticeable.

Maybe I should add a text then

**Reply by WorldLanguages** [reply]
I think we should preserve clear and reasonably sized buttons for "Scratch editor features" and "Scratch website features" that are always visible.

**Reply by Secret-chest** [reply]
> I think we should preserve clear and reasonably sized buttons for "Scratch editor features" and "Scratch website features" that are always visible.

So I should add a third pane with filters, maybe to the right of the sidenav?

**Reply by WorldLanguages** [reply]
@Secret-chest Not sure where it would fit

**Reply by Secret-chest** [reply]
> @Secret-chest Not sure where it would fit

@WorldLanguages if pinging you becomes a problem, tell me

Collapsible, but shown by default. There's quite a lot of free space

**Reply by WorldLanguages** [reply]
It could be collapsible, yeah

**Comment by BroJac5246** [comment]
Very nice! I would love to see separate "Hotkeys" and perhaps "Tweaks" sections, though that might be more specific to SA 2.0 in general.

**Reply by Secret-chest** [reply]
> Very nice! I would love to see separate "Hotkeys" and perhaps "Tweaks" sections, though that might be more specific to SA 2.0 in general.

Yes, a hotkey manager would make sense.

**Reply by Secret-chest** [reply]
What does Tweaks mean? Aren't they just addons?

**Reply by Secret-chest** [reply]
I also suggested a new grouping, type: fix, tweak, theme, addition

**Reply by BroJac5246** [reply]
> What does Tweaks mean? Aren't they just addons?

Yes, but it's a term I created for those small addons that tweak tiny things. This would allow us to add useful changes that might have been previously rejected or been considered borderline too-small but still added, such as Remove Search Bar Autocomplete or the proposed Compact Buttons addon. This could even include the Easter egg addon that capitalizes "Account settings."

**Reply by Secret-chest** [reply]
> > What does Tweaks mean? Aren't they just addons?
> 
> Yes, but it's a term I created for those small addons that tweak tiny things. This would allow us to add useful changes that might have been previously rejected or been considered borderline too-small but still added, such as Remove Search Bar Autocomplete or the proposed Compact Buttons addon. This could even include the Easter egg addon that capitalizes "Account settings."

It's pretty subjective. I'd argue compact buttons isn't a tweak.

**Reply by BroJac5246** [reply]
> > > What does Tweaks mean? Aren't they just addons?
> > 
> > Yes, but it's a term I created for those small addons that tweak tiny things. This would allow us to add useful changes that might have been previously rejected or been considered borderline too-small but still added, such as Remove Search Bar Autocomplete or the proposed Compact Buttons addon. This could even include the Easter egg addon that capitalizes "Account settings."
> 
> It's pretty subjective. I'd argue compact buttons isn't a tweak.

I could see a few different definitions for tweaks, it could be things that make small changes, or it could be modifications to vanilla Scratch behavior. The two are similar, but, as you pointed out, different.

**Comment by WorldLanguages** [comment]
Just a reminder that we plan to add setting descriptions to some settings. It would be great if you could consider that for your redesign.

I'm also wondering how this settings page would behave when the window is resized to closely match a mobile device's landscape mode. That "responsive" version of the settings page is pretty much the one used in the extension popup.

**Reply by Secret-chest** [reply]
Thanks, I'm doing it tomorrow.

It would go 1-column:
* Sidenav turns into hamburger menu.
* Addon cards would show the image above the description.
* Clicking on an addon card would bring up a modal with its settings.

**Reply by Secret-chest** [reply]
Just added all of your suggestions! @WorldLanguages

**Reply by Secret-chest** [reply]
I also added the messaging popup.

**Reply by WorldLanguages** [reply]
Nice to see you edited the post to consider these things!

**Reply by Secret-chest** [reply]
> Nice to see you edited the post to consider these things!

Thanks! Any thoughts on the popup UIs?

**Reply by WorldLanguages** [reply]
IMO the popup UI should stay more or less like it is now - all addons are collapsed by default (with no big images) and you can expand them to change the settings right from the popup.

**Reply by Secret-chest** [reply]
> IMO the popup UI should stay more or less like it is now - all addons are collapsed by default (with no big images) and you can expand them to change the settings right from the popup.

So basically the third view?

**Reply by Secret-chest** [reply]
What about the messaging @WorldLanguages ? It's a bit different. Also please ping me if you can, so I get notifications

**Reply by WorldLanguages** [reply]
@Secret-chest You should subscribe to your own forum thread.

**Reply by WorldLanguages** [reply]
> So basically the third view?

Yes, I assume in that view addons would continue to expand like they currently do, and when expanded it would look like the second view.

**Reply by Secret-chest** [reply]
> @Secret-chest You should subscribe to your own forum thread.

@WorldLanguages I'm already subscribed to the whole repo. If you mention me I also get mobile notifications.

**Comment by DNin01** [comment]
I think that layout makes it a lot easier to find addons you're looking for. Playing with those view options would be interesting, too.

**Reply by Secret-chest** [reply]
Thanks for the compliment.

**Reply by Secret-chest** [reply]
I now have 2 view options for desktop and 3 for mobile/popup.

**Comment by Samq64** [comment]
I know the mockup isn't really about the style, but please don't use that color sheme in the actual thing.

**Reply by WorldLanguages** [reply]
Of course not.
We *probably* aren't going to use material design UI elements either.

**Reply by Secret-chest** [reply]
> Of course not. We _probably_ aren't going to use material design UI elements either.

But maybe we could borrow some things, such as shadows instead of borders?

**Reply by Secret-chest** [reply]
Regarding the colour scheme, maybe we could get more colour modes, and this could be one of them.

**Reply by WorldLanguages** [reply]
I find it unlikely we'll ever add customizable color modes to our UIs.

**Reply by WorldLanguages** [reply]
> > Of course not. We _probably_ aren't going to use material design UI elements either.
> 
> But maybe we could borrow some things, such as shadows instead of borders?

I guess we could

**Comment by Secret-chest** [comment]
@WorldLanguages Would it be worthwhile to try it out in a fork (after the switch to Vue 3)?

**Reply by WorldLanguages** [reply]
I would suggest waiting until both Vue 3 and a cleanup are merged (they are done separately to avoid bugs). I can have this redesign in mind when cleaning up the settings code to possibly make this easier to implement.

**Reply by Secret-chest** [reply]
> I would suggest waiting until both Vue 3 and a cleanup are merged (they are done separately to avoid bugs). I can have this redesign in mind when cleaning up the settings code to possibly make this easier to implement.

I might need some Vue help since it's pretty hard. But JavaScript I already understand @WorldLanguages

**Reply by DNin01** [reply]
> > I would suggest waiting until both Vue 3 and a cleanup are merged (they are done separately to avoid bugs). I can have this redesign in mind when cleaning up the settings code to possibly make this easier to implement.
> 
> I might need some Vue help since it's pretty hard. But JavaScript I already understand @WorldLanguages 

Learning the Node.js commands needed for compilation might be a change, but it's not too bad.

**Comment by Secret-chest** [comment]
Is there any consensus?

**Reply by BroJac5246** [reply]
I don't think anyone is seriously against this. But I also don't think anyone is ready to say that they want this specific proposal since there's a few of them.

**Reply by Secret-chest** [reply]
So it's too early to decide?

**Reply by BroJac5246** [reply]
> So it's too early to decide?

There's currently no pressure to do so, so that appears to be the case. Once Scratch Addons 2.0 or some other major overhaul at some point in the future starts being made, that's when this is realistically likely to be discussed again (to my understanding).

**Reply by Samq64** [reply]
There's also the Vue 3 port that's kind of holding the settings page back.

**Reply by WorldLanguages** [reply]
Yeah, I find it likely we'll begin by porting the current settings page to Vue 3 and changing the font, then we can discuss another overhaul
